pkg/cmd/cli/cmd/examples.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+const examplesLong = `Displays one big list of examples for most commands available in the OpenShift client tools.
```

Comment:
Ok, makes sense. So I updated this PR to remove the '* examples' command and moved to a *separate* script similar to 'genbashcomp' (so it's *not* hidden under the end-user commands as I initially proposed). Looks cleaner. Notice this will be the basis for other docs generation that we may want to include after this, for example what's available upstream for 'kubectl'. @smarterclayton @liggitt PTAL

----- Original Message -----
> > @@ -0,0 +1,81 @@
> > +package cmd
> > +
> > +import (
> > +	"fmt"
> > +	"io"
> > +
> > +	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
> > +	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > +	"github.com/spf13/cobra"
> > +)
> > +
> > +const examplesLong = `Displays one big list of examples for most commands
> > available in the OpenShift client tools.
> 
> Ultimately my worry is the same as Jordan - I don't see it helping the
> average user.  A giant list of examples that can be found in our help docs
> (which is somewhat useful) is probably not something that is going to help a
> user succeed at their task.  For instance - is a user expected to read the
> whole thing?  Search it with grep?  If they know what to search for,
> wouldn't they also be likely to know what command they want (or just google
> it)?
> 
> > On Jun 25, 2015, at 4:49 PM, Fabiano Franz <notifications@github.com>
> > wrote:
> > 
> > In pkg/cmd/cli/cmd/examples.go:
> > 
> > > @@ -0,0 +1,81 @@
> > > +package cmd
> > > +
> > > +import (
> > > +	"fmt"
> > > +	"io"
> > > +
> > > +	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
> > > +	kcmdutil
> > > "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
> > > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > > +	"github.com/spf13/cobra"
> > > +)
> > > +
> > > +const examplesLong = `Displays one big list of examples for most
> > > commands available in the OpenShift client tools.
> > If the issue is having it *exposed* in help, one option is to remove the
> > description, like we do in `openshift infra`, which will remove it from
> > help everywhere.
> > …
> > —
> > Reply to this email directly or view it on GitHub.
> > 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/3325/files#r33314639